docs/guides/chip_tool_guide.md

```diff
@@ -0,0 +1,769 @@
+
+To build and run the `chip-tool`:
+
+1. Install all necessary packages and prepare the build system. For more
+   details, see the [Building Matter](BUILDING.md) documentation:
+
+    ```
+    sudo apt-get update
+    sudo apt-get upgrade
+
+    sudo apt-get install git gcc g++ python pkg-config libssl-dev libdbus-1-dev libglib2.0-dev libavahi-client-dev ninja-build python3-venv python3-dev python3-pip unzip libgirepository1.0-dev libcairo2-dev bluez
+    ```
+
+    If the `chip-tool` is built on a Raspberry Pi, install additional packages
+    and reboot the device:
+
+    ```
+    sudo apt-get install pi-bluetooth
+    sudo reboot
+    ```
+
+2. Clone the Project CHIP repository:
+
+    ```
+    git clone https://github.com/project-chip/connectedhomeip.git
+    ```
+
+3. Enter the `connectedhomeip` directory:
+
+    ```
+    cd connectedhomeip
+    ```
+
+4. Initialize the git submodules:
+
+    ```
+    git submodule update --init
+    ```
+
+5. Run the following command:
+
+    ```
+    ./scripts/examples/gn_build_example.sh examples/chip-tool BUILD_PATH
+    ```
+
+    `BUILD_PATH` specifies where the target binaries shall to be placed.
+
+6. To run the tool execute the following command:
+
+    ```
+    ./BUILD_PATH/chip-tool
+    ```
+
+    As a result, a list of available clusters will show up. Each cluster can be
+    used as a root of the new command corresponding to ZCL data model
+    interaction, for instance:
+
+    ```
+    ./BUILD_PATH/chip-tool pairing
+    ```
+
+    will print the list of sub-operations available for `pairing` cluster.
+
+    The following is one of the possible final commands which requires
+    particular command line arguments:
+
+    ```
+    ./BUILD_PATH/chip-tool pairing ble-thread
+    ```
+
+    Execution of above line will print an error and hint the proper list of
+    parameters which must be provided to run ZCL operation.
+
+<hr>
+
+<a name="using"></a>
+
+## Using `chip-tool` for Matter accessory testing
+
+This section describes how to use `chip-tool` to test the Matter accessory.
+Below steps depend on the application clusters that you implemented on the
+device side and may be different for your accessory.
+
+### Step 1: Prepare the Matter accessory.
+
+This tutorial is using the
+[Matter lighting-app example](https://github.com/project-chip/connectedhomeip/tree/master/examples/lighting-app)
+with the Bluetooth LE commissioning. However, you can adapt this procedure to
+other available Matter examples.
+
+Build and program the device with the Matter accessory firmware by following the
+example's documentation.
+
+### Step 2: Enable Bluetooth LE advertising on Matter accessory device.
+
+Some examples are configured to advertise automatically on boot. Other examples
+require physical trigger, for example pushing a button. Follow the documentation
+of the Matter accessory example to learn how Bluetooth LE advertising is enabled
+for the given example.
```

Comment:
Well, in this case you stated we're using the lighting app example, so I would expect clear information about how the advertising is enabled for the lighting app.

Reply:
My doubts here applies to the comment above ,regarding the *adapt* keyword :)